lib/request-worker.js

```diff
@@ -24,17 +24,15 @@ function doRequest() {
 	debug(`child request, cmd size: ${reqData.length} bytes`);
 	const req = JSON.parse(reqData);
 
-	if (!isSecurityEnhancedDomain(req.url)) {
-		process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
-	}
+	const agent = new https.Agent({ rejectUnauthorized: req.rejectUnauthorized || isSecurityEnhancedDomain(req.url) });
```

Comment:
You could avoid allocation of an HTTPS agent for a single operation and pass `rejectUnauthorized` (see https://nodejs.org/docs/v10.0.0/api/https.html#https_https_request_options_callback). This could allow also some other simplifications like merging it into options object.
Anyhow, I doubt anyone will notice the difference.